nc2zarr/writer.py

```diff
@@ -145,6 +169,10 @@ def _create_dataset(self, ds, encoding):
 
     def _append_dataset(self, ds: xr.Dataset):
         with log_duration('Appending dataset'):
+
+            if not self._dry_run:
```

Comment:
Done in commit 6c14d3b -- `_check_append_allowed` now also makes sure that the output Zarr exists before doing any checks on its data, since a dry run can involve simulating an append to a non-existent Zarr.